### ISSUE TITLE
timeuntil.t test may fail (race condition?)

### ISSUE THREAD

**Issue by eserte** [hurricup/DTL-Fast#59]
See http://www.cpantesters.org/cpan/report/39b95552-b38a-11e4-b221-9e126cbd7f71 for a sample fail report: 

```
#   Failed test 'Week and day'
#   at t/filter/timeuntil.t line 130.
#          got: '1 week, 23 hours'
#     expected: '1 week, 1 day'

#   Failed test 'Month and days'
#   at t/filter/timeuntil.t line 130.
#          got: '1 month, 1 day'
#     expected: '1 month, 2 days'

#   Failed test 'Year and day'
#   at t/filter/timeuntil.t line 130.
#          got: '1 year, 23 hours'
#     expected: '1 year, 1 day'
# Looks like you failed 3 tests of 22.
t/filter/timeuntil.t ........... 
```

This is with DTL-Fast-1.610, but I see this fail occasionally also with the newest 1.611.

**Comment by hurricup** [comment]
I've seen this bug and it's really strange. I don't understand what is going on and how. Sometimes, some systems shows these results but i can't reproduce them. Also, on the next build same systems may pass this test.

What is race condition in single-threaded system?

**Comment by eserte** [comment]
Well, this type of race condition could be that you call time() (or a moral equivalent) twice, and during this period the 2nd invocation of time() returned the next second. Just a possibility. Maybe it helps if you use the CPAN module `Time::Fake` to fake a specific time value.

I managed to create eight fail reports just by looping over the test suite for almost two hours.

**Comment by hurricup** [comment]
The test case looks like this:

{
        'template' => sprintf('{{ %s|timesince }}', time-24 \* 60 \* 60 \* 8 ),
        'test' => '1 week, 1 day',
        'title' => 'Week and day',
},

And time invoked only once per test, when data being prepared.
Hate such bugs...

**Comment by hurricup** [comment]
and i've never seen such bug on my dev machine (win+p5.20.1) or on my test machine (fbsd + perlbrew + perls from 10 to 20)

**Comment by eserte** [comment]
The race condition can be easily reproduced by adding a `sleep 1;` between the `$SET` definition and the `foreach` loop in `t/filter/timeuntil.t`.

**Comment by hurricup** [comment]
I don't really get your point.
We are defining `$SET`. After that all times been invoked already and sprintfed into templates as fixed numbers. No more time invocations. 
So any sleep before loop won't do anything.
Do I miss something?

**Comment by hurricup** [comment]
Oh crap... seems i'm stupid :D I've got it. thanks.
